### PR TITLE
[Joshua Seet] Duke Increments

### PR DESCRIPTION
**Pull request by joshuaseetss**

### CODE REVIEW

File: src/main/java/Duke.java [hunk 1]
```diff
@@ -49,16 +46,15 @@ public static void main(String[] args) throws IOException  {
...
     public void run() throws IOException {
-
         this.ui.welcome();
         String command = ui.promptEntry();
         ui.handleCommand(command, this.taskList);
         //after all commands are done, we will save the updated list into the txt file.
         ArrayList<Task> updated = this.taskList.getList();
 
-        if(!updated.isEmpty()) {
+        if (!updated.isEmpty()) {
```
Comment (lawncegoh) [inline src/main/java/Duke.java]: Genuine attempt in trying to comply with the coding standards. Good job Joshua!

File: src/main/java/TaskList.java [hunk 2]
```diff
@@ -45,7 +45,7 @@ public TaskList(ArrayList<Task> currentList) {
     public void addTask(Task current) {
```
Comment (lawncegoh) [inline src/main/java/TaskList.java]: The use of this TaskList might have been overly complicated, the sole purpose is to add and delete tasks and return the arraylist formed through the storage class

File: src/main/java/Duke.java [hunk 3]
```diff
@@ -123,53 +28,108 @@ public static String promptEntry() {
...
     }
 
-    public static void printList(ArrayList<Task> toDoList) {
-        int n = 1;
+    public void run() {
+
+        this.ui.welcome();
 
-        if(toDoList.isEmpty()){
+        String command = promptEntry();
```
Comment (lawncegoh) [inline src/main/java/Duke.java]: Further OOP can be achieved with the processing of commands shifted to a command class. Putting everything here in the run method() dissolves the purpose of OOP

File: src/main/java/Duke.java [hunk 4]
```diff
@@ -31,9 +34,40 @@ public static void main(String[] args) throws IOException  {
     }
 
     public static String promptEntry() {
-       return sc.next();
+        return sc.next();
     }
 
+    public static String getFormattedDate(String dateAndTime) {
```
Comment (lawncegoh) [inline src/main/java/Duke.java]: Good use of the SimpleDateFormat method from the Date Package, makes things a lot simpler

File: src/main/java/Duke.java [hunk 5]
```diff
@@ -14,154 +14,45 @@
...
         ArrayList<Task> existing = storage.readFileContents();
         this.taskList = new TaskList(existing);
         this.storage.writeToFile("");
 
-
     }
 
-
+    /**
+     * Runs the program. It will start from here.
```
Comment (lawncegoh) [inline src/main/java/Duke.java]: JavaDoc comments can be more detailed in explaining the methods

File: src/main/java/Duke.java [hunk 6]
```diff
@@ -31,9 +34,40 @@ public static void main(String[] args) throws IOException  {
     }
 
     public static String promptEntry() {
-       return sc.next();
+        return sc.next();
     }
 
+    public static String getFormattedDate(String dateAndTime) {
+        String result = dateAndTime;
```
Comment (uggi121) [inline src/main/java/Duke.java]: Convenient way of handling date-time inputs. +1!

File: src/main/java/Deadline.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    /**
+     * Recovery of a Deadline object based on the format of a task in the Duke.txt file that we save our information into.
+     * Whether the task is done or not depends on the int value of the task when it is saved.
+     *
+     * @param intDone if task is done, the int value == 1. If the task is not done, int value == 0.
+     * @param description of the task.
+     */
+    public Deadline(int intDone, String description, String date) {
```
Comment (uggi121) [inline src/main/java/Deadline.java]: Creative way of handling the isDone attribute. Another alternative may be using a setIsDone method

File: src/main/java/TaskList.java [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+     * @param current the Task object to be added to the list.
+     */
+    public void addTask(Task current) {
+
+        this.list.add(current);
+        if (list.size() > 1) {
+            System.out.println("Got it. I've added this task: \n" + "   " + current.toString() + "\n" +
+                    "Now you have " + list.size() + " tasks in the list. ");
+        } else {
+            System.out.println("Got it. I've added this task: \n" + "   " + current.toString() + "\n" +
```
Comment (uggi121) [inline src/main/java/TaskList.java]: Code could be a bit more OOP by delegating print commands to the UI class

File: src/main/java/UI.java [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+        return result;
+    }
+
+    /**
+     * Handles the commands entered by the user.
+     *
+     * @param command that the user entered.
+     * @param taskList the current tasklist, to be updated with every command.
+     */
+    public void handleCommand (String command, TaskList taskList) {
```
Comment (uggi121) [inline src/main/java/UI.java]: In my opinion this method is a bit too lengthy. One way is resolving this is to use a separate Parser class instead.

**Review by uggi121 — COMMENTED** [review]
Overall, I think your code is pretty good :) However, one change I'd definitely make is isolating the parser component from the UI class. Rest of it seems fine to me!

File: src/main/java/execution/Parser.java [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+    public static Command parse(String line) throws DukeException {
+        String[] split = line.split(" ");
+        String restOfInput = getRestOfInput(split);
+        return parseForCommand(split[0], restOfInput);
+
+    }
+
+    private static Command parseForCommand(String action, String rest) throws DukeException {
+        switch (action) {
+            case "bye":
```
Comment (yhtingg) [inline src/main/java/execution/Parser.java]: if I'm not wrong, I think the case and default expressions should be indented at the same level as the switch expression!

**Comment by yhtingg** [comment]
Your code is very concise and easy to read! Good job :)

**Pull request closed without merging**